java/vector/src/main/java/org/apache/arrow/vector/util/Validator.java

```diff
@@ -27,113 +27,123 @@
    */
-  public static void compareVectorSchemaRoot(VectorSchemaRoot root1, VectorSchemaRoot root2) {
+  public static boolean compareVectorSchemaRoot(VectorSchemaRoot root1, VectorSchemaRoot root2) {
     compareSchemas(root2.getSchema(), root1.getSchema());
     if (root1.getRowCount() != root2.getRowCount()) {
-      throw new IllegalArgumentException("Different row count:\n" + root1.getRowCount() + " != " + root2.getRowCount());
+      LOGGER.info("Different row count:\n" + root1.getRowCount() + " != " + root2.getRowCount());
+      return false;
     }
     List<FieldVector> vectors1 = root1.getFieldVectors();
     List<FieldVector> vectors2 = root2.getFieldVectors();
     if (vectors1.size() != vectors2.size()) {
-      throw new IllegalArgumentException("Different column count:\n" + vectors1.toString() +
-        "\n!=\n" + vectors2.toString());
+      LOGGER.info("Different column count:\n" + vectors1.toString() +
+          "\n!=\n" + vectors2.toString());
+      return false;
     }
     for (int i = 0; i < vectors1.size(); i++) {
-      compareFieldVectors(vectors1.get(i), vectors2.get(i));
+      if (!compareFieldVectors(vectors1.get(i), vectors2.get(i))) {
+        return false;
+      }
     }
+    return true;
   }
 
   /**
    * Validate two arrow FieldVectors are equal.
    *
    * @param vector1 the 1st VectorField to compare
    * @param vector2 the 2nd VectorField to compare
-   * @throws IllegalArgumentException if they are different
+   * @return true if FieldVectors are equal, otherwise false
    */
-  public static void compareFieldVectors(FieldVector vector1, FieldVector vector2) {
-    Field field1 = vector1.getField();
-    if (!field1.equals(vector2.getField())) {
-      throw new IllegalArgumentException("Different Fields:\n" + field1 + "\n!=\n" +
-        vector2.getField());
+  public static boolean compareFieldVectors(FieldVector vector1, FieldVector vector2) {
+    if (vector1.getClass() != vector2.getClass()) {
+      LOGGER.info("Different Fields:\n" + vector1.getField() + "\n!=\n" + vector2.getField());
+      return false;
     }
     int valueCount = vector1.getValueCount();
     if (valueCount != vector2.getValueCount()) {
-      throw new IllegalArgumentException("Different value count for field " + field1 + " : " +
-        valueCount + " != " + vector2.getValueCount());
+      LOGGER.info("Different valueCount:\n" + valueCount + "\n!=\n" + vector2.getValueCount());
+      return false;
     }
     for (int j = 0; j < valueCount; j++) {
       Object obj1 = vector1.getObject(j);
       Object obj2 = vector2.getObject(j);
-      if (!equals(field1.getType(), obj1, obj2)) {
-        throw new IllegalArgumentException(
-            "Different values in column:\n" + field1 + " at index " + j + ": " + obj1 + " != " + obj2);
+      if (!equals(vector1.getField().getType(), obj1, obj2)) {
+        LOGGER.info("Different values in column:\n" + vector1.getField() +
```

Comment:
If we want to keep original error message, it might not proper to refactor Validator to make it has return type and use it in Dictionary, just copy it logic to Dictionary as previous commit does? Or any good suggestions? Thanks!